app/src/org/commcare/android/util/AndroidUtil.java

```diff
@@ -71,4 +79,51 @@ public static void initializeStaticHandlers() {
             return a;
         }
     }
+
+    public static void setClickListenersForEverything(final Activity activity, final ViewGroup v) {
+        if (BuildConfig.DEBUG) {
+            final ViewGroup layout = v != null ? v : (ViewGroup) activity.findViewById(android.R.id.content);
+            final LinkedList<View> views = new LinkedList<View>();
+            views.add(layout);
+            for (int i = 0; !views.isEmpty(); i++) {
+                final View child = views.getFirst();
+                views.removeFirst();
+                Log.i("GetID", "Adding onClickListener to view " + child);
+                child.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(final View v) {
+                        String vid;
+                        try {
+                            vid = "View id is: " + v.getResources().getResourceName(v.getId()) + " ( " + v.getId() + " )";
+                        } catch (final Resources.NotFoundException excp) {
+                            vid = "View id is: " + v.getId();
+                        }
+                        Log.i("CLK", vid);
+                    }
+                });
+                if(child instanceof ViewGroup) {
+                    final ViewGroup vg = (ViewGroup) child;
+                    for (int j = 0; j < vg.getChildCount(); j++) {
+                        final View gchild = vg.getChildAt(j);
+                        if (!views.contains(gchild)) views.add(gchild);
+                    }
+                }
+            }
+        }
+    }
+
+    public static void setClickListenersForEverything(final Activity act){
+        setClickListenersForEverything(act, (ViewGroup) act.findViewById(android.R.id.content));
+    }
+
+    public static int[] getThemeColorIDs(final Context context, final int[] attrs){
```

Comment:
Comment on what this method does (specifically what it takes as arguments and returns, since int arrays are very non-descriptive)